### ISSUE TITLE
meteor 1.4.1.2 build - windows 10 x64, npm install on bundle/programs/server gives error 

### ISSUE THREAD

**Issue by oberoivarun** [meteor/meteor#7900]
This is the angular2-meteor-base --example application. When i run `meteor` the app runs fine on the browser.

I am making a small meteor app to deploy in a local environment with about 6-7 users and so i used meteor build to create a nodejs app.

Instructions in the produced bundle say:

```
This is a Meteor application bundle. 
It has only one external dependency:
Node.js 4.4.7 or newer. 
To run the application:
$ (cd programs/server && npm install)

$ export MONGO_URL='mongodb://user:password@host:port/databasename'

$ export ROOT_URL='http://example.com'

$ export MAIL_URL='smtp://user:password@mailhost:port/'

$ node main.js

Use the PORT environment variable to set the port where the
application will listen. 
The default is 80, but that will require
root on most systems.


Find out more about Meteor at meteor.com.
```

The error happens when i do `npm install` in the server directory:

Error:

```
fibers@1.0.13 install C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers
> node build.js || nodejs build.js


C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild --release )  else (node "" rebuild --release )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  fibers.cc
  coroutine.cc
..\src\fibers.cc(727): warning C4244: 'argument': conversion from 'std::size_t' to 'double', possible loss of data [
C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers\build\fibers.vcxproj]
..\src\fibers.cc(731): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data [C:\User
s\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers\build\fibers.vcxproj]
..\src\fibers.cc(738): warning C4244: 'argument': conversion from 'std::size_t' to 'double', possible loss of data [
C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers\build\fibers.vcxproj]
..\src\coroutine.cc(137): warning C4552: '!': operator has no effect; expected operator with side-effect [C:\Users\v
arun\Desktop\bundle\bundle\programs\server\node_modules\fibers\build\fibers.vcxproj]
..\src\coroutine.cc(214): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of da
ta [C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers\build\fibers.vcxproj]
  coro.c
  win_delay_load_hook.c
C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\src\win_delay_load_hook.c(34): error C2373: '__pfnDli
NotifyHook2': redefinition; different type modifiers [C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modu
les\fibers\build\fibers.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\delayimp.h(134): note: see declaration of '__pfnDli
  NotifyHook2'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--release"
gyp ERR! cwd C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers
gyp ERR! node -v v6.7.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++`
Alpine users please run: `sudo apk add python make g++`
'nodejs' is not recognized as an internal or external command,
operable program or batch file.
npm WARN meteor-dev-bundle@0.0.0 No description
npm WARN meteor-dev-bundle@0.0.0 No repository field.
npm WARN meteor-dev-bundle@0.0.0 No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE

npm ERR! fibers@1.0.13 install: `node build.js || nodejs build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fibers@1.0.13 install script 'node build.js || nodejs build.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the fibers package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build.js || nodejs build.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs fibers
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls fibers
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\varun\Desktop\bundle\bundle\programs\server\npm-debug.log
```

Updated Node, updated windows build tools, ran into different problems, so installed node, python, VS code, many different libraries, searched the error a lot and tried many different solutions, still stuck. Please suggest.

**Comment by abernix** [comment]
Please try using Node version 4 (NOT Node 6) and see what your results are.  The message which says "Node.js 4.4.7 or newer" is meant to indicate newer along the 4.x branch of Node.  I realize this isn't 100% clear and I'll see about getting that fixed.

**Comment by oberoivarun** [comment]
With node 4.6.0, this is the error i receive. Again, related to the fibers package.

I also tried npm init on an empty directory and then tried ```npm install fibers@1.0.13```... It gave me a similar error.

instead when i did ```npm install fibers``` without any version option, it installed the package without error.

```
C:\Users\varun\Desktop\bundle\bundle\programs\server>npm install
npm WARN package.json meteor-dev-bundle@0.0.0 No description
npm WARN package.json meteor-dev-bundle@0.0.0 No repository field.
npm WARN package.json meteor-dev-bundle@0.0.0 No README data
npm WARN package.json meteor-dev-bundle@0.0.0 No license field.
\
> fibers@1.0.13 install C:\Users\varun\Desktop\bundle\bundle\programs\server\node_modules\fibers
> node build.js || nodejs build.js

`win32-x64-v8-4.5` exists; testing
Binary is fine; exiting

> meteor-dev-bundle@0.0.0 install C:\Users\varun\Desktop\bundle\bundle\programs\server
> node npm-rebuild.js

events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: spawn npm ENOENT
    at exports._errnoException (util.js:907:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
    at Function.Module.runMain (module.js:443:11)
    at startup (node.js:139:18)
    at node.js:974:3

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v4.6.0
npm ERR! npm  v2.15.9
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 install: `node npm-rebuild.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the meteor-dev-bundle@0.0.0 install script 'node npm-rebuild.js'.
npm ERR! This is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node npm-rebuild.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\varun\Desktop\bundle\bundle\programs\server\npm-debug.log
```

**Comment by staskorz** [comment]
@oberoivarun , if you deploy to the same platform you developed on (say both are Windows x64), you could try adding the following environment variable _at machine level_ before running `npm install`:
`METEOR_SKIP_NPM_REBUILD=1`

**Comment by abernix** [comment]
This came up again in #8007.  It looks like this is a duplicate of that and will be fixed with 19fdc5c65faa28834f26403fb6ad2d7984e2c1f4, thanks to @benjamn's commit.  Thanks for reporting this and sorry it took a bit to resolve!

This is included in the `1.4.2.1-rc.0` release candidate right now so if you want to try it out, please:

```
cd /your/meteor/app
meteor update --release 1.4.2.1-rc.0
# proceed as usual!
```

Closing but happy to reopen (or you can comment in the other issue, preferably),  if the problem continues!
